src/ui/main_window.py

```
@@ -317,3 +335,96 @@ def autosave_experiment(self) -> None:
+            QMessageBox.warning(
+                self, "Invalid ROI", "Current ROI is not valid for cropping."
+            )
+            return
+
+        # Ellipse ROI
+        roi_tuple = self.viewer.current_roi
+        # Validate ROI tuple contains numeric values (works even with -O flag)
+        if not isinstance(roi_tuple, (tuple, list)) or len(roi_tuple) != 4:
+            raise ValueError(
+                f"ROI must be a 4-element tuple or list, got {type(roi_tuple)}"
+            )
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)